core_java/twitter/src/main/java/ca/jrvs/apps/twitter/DAO/TwitterHttpHelper.java

```diff
@@ -0,0 +1,121 @@
+package ca.jrvs.apps.twitter.DAO;
+
+import com.google.gdata.util.common.base.PercentEscaper;
+import java.io.IOException;
+import java.net.URI;
+import java.sql.Timestamp;
+import oauth.signpost.OAuthConsumer;
+import oauth.signpost.commonshttp.CommonsHttpOAuthConsumer;
+import oauth.signpost.exception.OAuthCommunicationException;
+import oauth.signpost.exception.OAuthExpectationFailedException;
+import oauth.signpost.exception.OAuthMessageSignerException;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+
+public class TwitterHttpHelper implements HttpHelper {
+  private static String CONSUMER_KEY;
+  private static String CONSUMER_SECRET;
+  private static String ACCESS_TOKEN;
+  private static String TOKEN_SECRET;
+
+  private OAuthConsumer consumer;
+  private HttpClient httpClient;
+
+  public TwitterHttpHelper(String consumerKey, String consumerSecret, String accessToken, String tokenSecret) {
+    this.CONSUMER_KEY = consumerKey;
+    this.CONSUMER_SECRET = consumerSecret;
+    this.ACCESS_TOKEN = accessToken;
+    this.TOKEN_SECRET = tokenSecret;
+  }
+
+  public HttpResponse httpIntermediate(boolean type, URI uri)
```

Comment:
You should use an Enum instead of a boolean here if using this approach. `httpIntermediate(true/false, myURI);` is ambiguous.

You're also repeating yourself in this function, consider moving the `new HttpGet()` and `new HttpPost()` into the functions that call this since you wont need to check a flag to determine if it's a GET or a POST request in those, meaning you can just sign an execute.